Source/Parallelization/WarpXComm.cpp

```diff
@@ -181,7 +181,9 @@ WarpX::UpdateAuxilaryDataStagToNodal ()
             // ParallelCopy from coarse level
             for (int i = 0; i < 3; ++i) {
                 IntVect ng = Btmp[i]->nGrowVect();
-                Btmp[i]->ParallelCopy(*Bfield_aux[lev-1][i], 0, 0, 1, ng, ng, cperiod);
+                // Guard cells may not be up to date beyond ng_FieldGather
+                const auto ng_src = guard_cells.ng_FieldGather;
+                Btmp[i]->ParallelCopy(*Bfield_aux[lev-1][i], 0, 0, 1, ng_src, ng, cperiod);
```

Comment:
I wonder if an additional inline comment on the different role of the two arguments `ng_src` and `ng` passed to `ParallelCopy` (as well as the `0`s and `1`s maybe...) could help understand this function call better when going through the code. Same comment for the similar calls below.

Reply:
I've added an additional comment describing the role of `ng` and `ng_src`. I didn't describe the 0s and 1s which are for the MultiFAB components and number of components, because I feel it would be a bit burdensome to explicitly describe them everytime we call MultiFAB functions (especially since here we don't use these parameters in a tricky/unusual way, unlike for the guard cell numbers).

Let me know if that's ok for you.